### PR TITLE
Fixes #23888 - support enable/disable recurring logic

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by iNecas** [comment]
So far looks good: have not tested yet. Mind covering this with some unit tests?

**Comment by jlsherrill** [comment]
```
having frozen tasks marked in the UI
```
Is the best way to do this be to go through its task groups to lookup its recurring logic?  It seems like a task could have multiple (although i'm not sure thats possible).  Is it safe to assume one?

**Comment by adamruzicka** [comment]
Yeah, just looking at the delayed plan should be enough

File: app/models/foreman_tasks/task/dynflow_task.rb [hunk 1]
```diff
@@ -65,6 +65,14 @@ def execution_plan(silence_exception = true)
       nil
     end
 
+    def frozen
+      delayed_plan.try(:frozen)
+    end
+
+    def delayed_plan
+      ForemanTasks.dynflow.world.persistence.load_delayed_plan(external_id)
```
Comment (adamruzicka) [inline app/models/foreman_tasks/task/dynflow_task.rb]: How about returning nil early if the task is not in scheduled state so we won't hit the db 20 times each time a user opens the task index just to see that there are no delayed plans for the tasks?

File: app/helpers/foreman_tasks/foreman_tasks_helper.rb [hunk 2]
```diff
@@ -52,14 +55,16 @@ def duration_in_words_span(start, finish)
...
+        buttons << link_to(N_('Cancel'), cancel_foreman_tasks_recurring_logic_path(recurring_logic), :method => :post, :class => '') unless recurring_logic.done?
       end
-      button_group buttons
+      action_buttons buttons
     end
 
     def recurring_logic_next_occurrence(recurring_logic)
       default = '-'
-      return default if %w[finished cancelled].include? recurring_logic.state
+      return default if recurring_logic.done? || !recurring_logic.enabled?
```
Comment (adamruzicka) [inline app/helpers/foreman_tasks/foreman_tasks_helper.rb]: Why not use `recurring_logic.disabled?`

File: app/controllers/foreman_tasks/recurring_logics_controller.rb [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
     def show; end
 
+    def enable
+      @recurring_logic.update!(:enabled => true)
+      redirect_to :action => :index
+    end
+
+    def disable
+      @recurring_logic.update!(:enabled => false)
+      redirect_to :action => :index
```
Comment (adamruzicka) [inline app/controllers/foreman_tasks/recurring_logics_controller.rb]: Clicking disable/enable when looking at recurring logic details redirects you to the index. I think the it should stay on the same page, but on the other hand cancel behaves the same way. Guess we can be consistent first and fix this behavior in another PR

**Comment by jlsherrill** [comment]
@adamruzicka fixed the nitpicks and added an error when enabling a cancelled task.

**Comment by jlsherrill** [comment]
Ah right, good call, updated!

**Comment by adamruzicka** [comment]
Dynflow 1.1.0 with the needed changes was just released, please bump the dependency here

**Comment by jlsherrill** [comment]
[test foreman-tasks]

**Comment by jlsherrill** [comment]
These failures look unrelated?   I can reproduce locally but even on master.

**Comment by iNecas** [comment]
[test foreman-tasks] the failures should be fixed now with https://github.com/theforeman/foreman/pull/5823

**Comment by jlsherrill** [comment]
@iNecas Thanks! its :green_heart:

**Comment by iNecas** [comment]
@adamruzicka no other comments from me

**Comment by adamruzicka** [comment]
Thanks @jlsherrill !

**Pull request merged**